### PR TITLE
ROIPool + Dispatcher + Autocast + Code Cleanup

### PR DESCRIPTION
**Pull request by datumbox**
part of #2797 and #2796

Implements Dispatcher + Autocast for ROIPool.

I'm also doing some clean up on the original code such as:
- Remove primitive const declaration from method names.
- Using references when possible.

### CODE REVIEW

File: torchvision/csrc/ROIPool.h [hunk 1]
```diff
@@ -117,13 +126,77 @@ class ROIPoolFunction : public torch::autograd::Function<ROIPoolFunction> {
...
+      rois,
+      argmax,
+      spatial_scale,
+      pooled_height,
+      pooled_width,
+      batch_size,
+      channels,
+      height,
+      width)[0];
+}
```
Comment (fmassa) [inline torchvision/csrc/ROIPool.h:202]: nit: newline

**Review by fmassa — COMMENTED** [review]
Code looks great, thanks!

Can you add some tests for autocast as well?
We have an example of it for RoIAlign in https://github.com/pytorch/vision/blob/455cd57c9404429448280cd4a0d7f600a51afc54/test/test_ops.py#L298-L303

File: torchvision/csrc/ROIPool.h (hunk 1 above)
Comment (datumbox) [reply torchvision/csrc/ROIPool.h:202]: done

**Comment by fmassa** [comment]
Discussed with @datumbox offline, in order to avoid conflicts with some currently ongoing PRs he will be sending the tests in a separate PR

**Review by fmassa — APPROVED** [review]
Thanks a lot!

**Pull request merged**